packages/expo/src/Linking/Linking.ts

```diff
@@ -129,12 +129,14 @@ function parse(url: string): ParsedURL {
     let expoPrefix: string | null = null;
     if (hostUriStripped) {
       const parts = hostUriStripped.split('/');
-      expoPrefix = `${parts.slice(1).join('/')}/--/`;
+      expoPrefix = `${parts
+        .slice(1)
+        .concat(['--/'])
+        .join('/')}`;
     }
 
     if (IS_EXPO_HOSTED && !USES_CUSTOM_SCHEME && expoPrefix && path.startsWith(expoPrefix)) {
       path = path.substring(expoPrefix.length);
-      hostname = null;
```

Comment:
This might have been there intentionally so we keep the hostname for Expo-hosted assets abstract for now. (In the future we want to treat all hostnames the same but that’s a bigger change).

Reply:
yeah good point, adding it back in.

This API is a little confusing right now, since it does quite a bit of manipulation of the url if it's in the Expo client. Seems like it might be better to just treat it as a regular url, and provide the full path, host, etc? But I'm assuming it's like this `hostname = null` case, where it was done intentionally for a good reason and I'm just failing to see it 😄